typeDefs.js

```diff
@@ -560,6 +560,7 @@ const typeDefs = gql`
       filter: MaterialFilterInput,
     ): PaginatedMaterials!
     getPromoCodeById(id: ID): PromoCode
+    getPromoCodeByCode(code: String!): PromoCode
```

Comment:
please attach here how PromoCode fields looks

Reply:
if in all promocodes methods error case not handled, need to create task for refactoring

Reply:
In progress

Reply:
Done